agent/org.linkedin.glu.agent-server-upgrade/src/cmdline/resources/bin/agentctl.sh

```diff
@@ -202,14 +204,19 @@ get_pid()
     PID=`cat $PID_FILE`
     PS_STAT=`ps -p $PID -o'user,pid=' | tail -1 | awk '{print $2}'`
 
-    case "$PS_STAT" in
-      $PID  ) echo $PID
-              ;;
-      *     ) echo 0
-              ;;
-    esac
+    if [[ "$PS_STAT" -eq "$PID" ]]; then
+      CMD=$(ps -p $PID -o 'comm=')
```

Comment:
Sure, I'm using ps to find the name of the program for PID $PID which we got from the PID_FILE.

$ cat /var/glu/data/logs/org.linkedin.glu.agent-server.pid
29302
$ ps -p 29302 -o 'comm='
java

I then do a string comparison to see if the command I found is equal to the command I expect to run with $JAVA_CMD.  In my case $JAVA_CMD contains the full path to java for example JAVA_CMD=/opt/local/java/bin/java so basename removes all leading directory components.

$ basename /opt/local/java/bin/java
java

If you are curious basename works even with relative paths or if there is no leading directories.

$ basename java
java
$ basename ../bin/java
java

I suspect $(basename "$JAVA_CMD") will always return "java", but maybe someone out there is using something like Android's Dalvik so I figure it is best to find the command from JAVA_CMD.

This is all just a basic sanity check.  We could still find a process with the PID we are looking for that happens to be a java process, but is not the glu-agent.  However that is much more unlikely than the situation without the check.